### PR TITLE
CORE-7943 Use separate SerializationOutput and DeserializationInput to have more realistic serialization tests

### PR DESCRIPTION
**Pull request by vlajos**
CORE-7943 Use separate SerializationOutput and DeserializationInput to have more realistic serialization tests

### CODE REVIEW

File: testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt [hunk 1]
```diff
@@ -67,25 +70,20 @@ abstract class CommonLedgerIntegrationTest {
...
+        wireTransactionFactory = sandboxGroupContext.getSandboxSingletonService()
+        kryoSerializer = sandboxGroupContext.getObjectByKey(FlowSandboxGroupContextImpl.CHECKPOINT_SERIALIZER)
             ?: fail("No CheckpointSerializer in sandbox context")
-        sandboxSerializationService1 = sandboxGroupContext1.getObjectByKey(RequireSandboxAMQP.AMQP_SERIALIZATION_SERVICE)
-            ?: fail("No Serializer in sandbox context")
-        sandboxSerializationService2 = sandboxGroupContext2.getObjectByKey(RequireSandboxAMQP.AMQP_SERIALIZATION_SERVICE)
-            ?: fail("No Serializer in sandbox context")
+        serializationOutput = SerializationOutput(sandboxGroupContext.createSerializerFactory())
+        deserializationInput = DeserializationInput(sandboxGroupContext.createSerializerFactory())
+        serializationContext = AMQP_STORAGE_CONTEXT.withSandboxGroup(sandboxGroupContext.sandboxGroup)
```
Comment (chrisr3) [inline testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt]: Looking at this, I'm thinking we only really need something like:
```kotlin
lateinit var serializationService: SerializationService
```
followed by
```kotlin
serializationService = SerializationServiceImpl(
    serializationOutput = SerializationOutput(sandboxGroupContext.createSerializerFactory()),
    deserializationInput = DeserializationInput(sandboxGroupContext.createSerializerFactory()),
    context = AMQP_STORAGE_CONTEXT.withSandboxGroup(sandboxGroupContext.sandboxGroup)
)
```
And accompanied by a suitable comment too, explaining why we haven't used the sandbox's default `serializationService`.

The tests themselves would then need very few changes.

File: testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt [hunk 2]
```diff
@@ -40,16 +43,16 @@ abstract class CommonLedgerIntegrationTest {
...
-    val testSerializationContext = AMQP_STORAGE_CONTEXT
-
-    lateinit var flowSandboxService: FlowSandboxService
-    lateinit var sandboxGroupContext1: SandboxGroupContext
+    private lateinit var flowSandboxService: FlowSandboxService
+    lateinit var sandboxGroupContext: SandboxGroupContext
     lateinit var jsonMarshallingService: JsonMarshallingService
-    lateinit var sandboxSerializationService1: SerializationService
-    lateinit var sandboxSerializationService2: SerializationService
+    lateinit var serializationFactory: SerializationFactory
```
Comment (chrisr3) [inline testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt]: This `serializationFactory` property looks unused.

**Review by chrisr3 — COMMENTED** [review]
Nice, but I think we can simplify even further.

File: testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt (hunk 1 above)
Comment (vlajos) [reply testing/ledger/ledger-common-base-integration-test/src/main/kotlin/net/corda/ledger/common/integration/test/CommonLedgerIntegrationTest.kt]: Thanks. I have updated it.

**Review by chrisr3 — APPROVED** [review]
LGTM

**Review by mcgovc — APPROVED** [review]
LGTM - build changes

**Pull request merged**